src/test/java/org/fogbowcloud/manager/core/models/linkedlist/SynchronizedDoublyLinkedListTest.java

```diff
@@ -0,0 +1,242 @@
+package org.fogbowcloud.manager.core.models.linkedlist;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.fail;
+
+import org.fogbowcloud.manager.core.models.orders.ComputeOrder;
+import org.fogbowcloud.manager.core.models.orders.Order;
+import org.junit.Before;
+import org.junit.Test;
+
+public class SynchronizedDoublyLinkedListTest {
+
+	private SynchronizedDoublyLinkedList list;
+	
+	@Before
+	public void initialize() {
+		this.list = new SynchronizedDoublyLinkedList();
+	}
+	
+	@Test
+	public void testAddFirst() {
+		assertNull(this.list.getHead());
+		assertNull(this.list.getTail());
+		
+		Order order = createOrder("0");
+		this.list.addItem(order);
+
+		assertEquals(order, this.list.getHead().getOrder());
+		assertEquals(order, this.list.getCurrent().getOrder());
+		assertEquals(order, this.list.getTail().getOrder());
+	}
+	
+	@Test
+	public void testAddOrder() {
+		assertNull(this.list.getNext());
+		
+		Order orderOne = createOrder("one");
+		Order orderTwo = createOrder("two");
+		this.list.addItem(orderOne);
+		this.list.addItem(orderTwo);
+		
+		assertEquals(orderOne, this.list.getNext());
+		assertEquals(orderTwo, this.list.getNext());
+		// Next node of the second order is null.
+		assertNull(this.list.getNext());
+		
+		Order orderThree = createOrder("three");
+		this.list.addItem(orderThree);
+
+		// for now the pointer is on null.
+		assertEquals(orderThree, this.list.getNext());
+	}
+	
+	@Test
+	public void testAddNullOrder() {
+		assertNull(this.list.getNext());
+		
+		Order orderNull = null;
+		try {
+			this.list.addItem(orderNull);
+			fail("Null order should not be added.");
+		} catch (IllegalArgumentException e){
+			assertEquals("Attempting to add a null order.", e.getMessage());
+		}
+	}
+	
+	@Test
+	public void testResetPointer() {
+		assertNull(this.list.getNext());
```

Comment:
Same comment as before. Use assertNull(this.list.getHead());

However, this test seems to be redundant considering the fix suggested in testAddOrder().